widgetssdk/src/main/java/com/glia/widgets/chat/ChatActivity.kt

```diff
@@ -0,0 +1,186 @@
+        )
+    }
+
+    override fun onResume() {
+        chatView.onResume()
+        super.onResume()
+    }
+
+    override fun onPause() {
+        super.onPause()
+        chatView.onPause()
+    }
+
+    override fun onDestroy() {
+        chatView.onDestroyView()
+        Logger.i(TAG, "Destroy Chat screen")
+        super.onDestroy()
+    }
+
+    private fun createEngagementConfiguration(intent: Intent): EngagementConfiguration = EngagementConfiguration(intent)
+
+    //TODO: Check why theme attribute is not anymore used
+    private fun startCallScreen(theme: UiTheme, mediaType: String) {
+        startActivity(
+            CallActivity.getIntent(
+                applicationContext,
+                defaultCallConfiguration.copy(mediaType = Utils.toMediaType(mediaType), isUpgradeToCall = true)
+            )
+        )
+        finish()
+    }
+
+    private fun backToCallScreen() {
+        startActivity(CallActivity.getIntent(applicationContext, defaultCallConfiguration))
+        finish()
+    }
+
+    /**
+     * Just commenting to avoid lint error, this should be removed in next tasks
+     */
+    companion object {
+        private val TAG: String = ChatActivity::class.java.simpleName
+
+        /**
+         * Creates and fills out Intent for starting ChatActivity
+         *
+         * @param context   - Android Context object
+         * @param contextId - Glia visitor context asset ID
+         * @param queueIds  - Queue IDs of the queues you would like to use for your engagements
+         * @return - Intent for Starting ChatActivity
+         */
+        fun getIntent(
+            context: Context,
+            contextId: String?,
+            queueIds: List<String>
+        ): Intent {
+            return Intent(context, ChatActivity::class.java)
+                .putExtra(GliaWidgets.CONTEXT_ASSET_ID, contextId)
+                .putStringArrayListExtra(GliaWidgets.QUEUE_IDS, ArrayList(queueIds))
+        }
+
+        /**
+         * Creates and fills out Intent for starting ChatActivity
+         *
+         * @param context   - Android Context object
+         * @param contextId - Glia visitor context asset ID
+         * @param queueIds  - Queue IDs of the queues you would like to use for your engagements
+         * @param chatType  - Type of chat screen
+         * @return - Intent for Starting ChatActivity
+         */
+        fun getIntent(
+            context: Context,
+            contextId: String?,
+            queueIds: List<String>,
+            chatType: ChatType?
+        ): Intent {
+            return Intent(context, ChatActivity::class.java)
+                .putExtra(GliaWidgets.CONTEXT_ASSET_ID, contextId)
+                .putExtra(GliaWidgets.QUEUE_IDS, ArrayList(queueIds))
+                .putExtra(GliaWidgets.COMPANY_NAME, "Legacy company")
+                .putExtra(GliaWidgets.CHAT_TYPE, chatType as Parcelable?)
+        }
```

Comment:
I know the is clearly out of the scope of this PR, but could you extract this into `CallActivityIntentHelper` class and then rename that class to something like `WidgetsActivityIntentHelper`?
But ignore this suggestion if you are busy with something else or it ends up as bigger refactoring

Reply:
This and other Intent-related things will be refactored in the next PR.

Reply:
It definitely makes sense to have Intent creation functions in one place. What I think is confusing - we have `CallActivityIntentHelper` but don't have `ChatActivityIntentHelper`.